src/Utils.cpp

```diff
@@ -499,6 +499,82 @@ CString CUtils::FormatTime(time_t t, const CString& sFormat,
     return s;
 }
 
+/* Supports additional format specifiers for formatting sub-second values:
+ *
+ * - %L - millisecond, 3 digits (ruby extension)
+ * - %N - sub-second fraction (ruby extension)
+ *   - %3N - millisecond
+ *   - %6N - microsecond
+ *   - %9N - nanosecond (default if no digit specified)
+ *
+ * However, note that timeval only supports microsecond precision
+ * (thus, formatting with higher-than-microsecond precision will
+ * always result in trailing zeroes), and IRC server-time is specified
+ * in millisecond precision (thus formatting received timestamps with
+ * higher-than-millisecond precision will always result in trailing
+ * zeroes).
+ */
+CString CUtils::FormatTime(timeval tv, const CString& sFormat,
+                           const CString& sTimezone) {
+    // Parse additional format specifiers before passing them to
+    // strftime, since the way strftime treats unknown format
+    // specifiers is undefined.
+    CString sFormat2;
```

Comment:
I had that in my initial implementation, then removed it. Reserving memory here would mean we're always doing at least one allocation (assuming small-string optimization doesn't kick in of course), which we're trying to avoid if it's not necessary.